### PR TITLE
fix: remove short flag -j for json and return response to global json handler

### PR DESCRIPTION
**Pull request by RasPhilCo**
### What does this PR do?
Remove -j from json flag and moves --json into global flags in help

Use this.warn to add warnings to json object

Return result from run for json handler & schema check

### What issues does this PR fix or reference?
@W-00000000@

### CODE REVIEW

**Review by saracope — APPROVED** [review]

File: src/commands/deploy/functions.ts [hunk 1]
```diff
@@ -190,13 +190,7 @@ export default class DeployFunctions extends Command {
...
     if (flags.json) {
-      cli.styledJSON({
-        status: 0,
-        result: {
-          results,
-        },
-        warnings: [],
-      });
+      return results;
     }
```
Comment (RodEsp) [inline src/commands/deploy/functions.ts:194]: You don't need, and in fact you don't want, the `if (flags.json)` here. Just do `return results;` without an `if` statement. These don't get displayed in the terminal output unless the user passes in `--json` but they _do_ need to be returned always for the command schema generation.

File: src/commands/env/create/compute.ts [hunk 2]
```diff
@@ -181,17 +181,13 @@ export default class EnvCreateCompute extends Command {
     }
     const app = await fetchAppForProject(this.client, projectName, org.getUsername());
     if (flags.json) {
```
Comment (RodEsp) [inline src/commands/env/create/compute.ts:183]: Same, no `if` statement needed.

File: src/commands/env/delete.ts [hunk 3]
```diff
@@ -158,11 +158,7 @@ export default class EnvDelete extends Command {
     cli.action.stop();
 
     if (flags.json) {
```
Comment (RodEsp) [inline src/commands/env/delete.ts:160]: Same, no `if` statement needed.

File: src/commands/env/logdrain/add.ts [hunk 4]
```diff
@@ -86,20 +86,16 @@ export default class LogDrainAdd extends Command {
       });
 
       if (flags.json) {
```
Comment (RodEsp) [inline src/commands/env/logdrain/add.ts:88]: Same, no `if` statement needed.

File: src/commands/env/logdrain/list.ts [hunk 5]
```diff
@@ -57,43 +57,32 @@ export default class LogDrainList extends Command {
 
     if (flags.json) {
```
Comment (RodEsp) [inline src/commands/env/logdrain/list.ts:58]: Same, no `if` statement needed.

File: src/commands/env/logdrain/remove.ts [hunk 6]
```diff
@@ -77,12 +77,7 @@ export default class LogDrainRemove extends Command {
     try {
       await this.client.delete<Heroku.LogDrain>(`/apps/${appName}/log-drains/${encodeURIComponent(url)}`);
       if (flags.json) {
```
Comment (RodEsp) [inline src/commands/env/logdrain/remove.ts:79]: Same, no `if` statement needed.

File: src/commands/env/var/get.ts [hunk 7]
```diff
@@ -69,28 +69,21 @@ export default class VarGet extends Command {
 
     if (flags.json) {
```
Comment (RodEsp) [inline src/commands/env/var/get.ts:70]: Same, no `if` statement needed.

File: src/commands/env/var/list.ts [hunk 8]
```diff
@@ -66,43 +66,33 @@ export default class ConfigList extends Command {
     });
 
     if (flags.json) {
```
Comment (RodEsp) [inline src/commands/env/var/list.ts:68]: Same, no `if` statement needed.

File: src/commands/env/var/unset.ts [hunk 9]
```diff
@@ -106,11 +106,7 @@ export default class ConfigUnset extends Command {
     }
 
     if (flags.json) {
```
Comment (RodEsp) [inline src/commands/env/var/unset.ts:108]: Same, no `if` statement needed.

File: src/commands/login/functions/jwt.ts [hunk 10]
```diff
@@ -209,21 +208,17 @@ export default class JwtLogin extends Command {
 
     await this.stateAggregator.tokens.write();
 
+    cli.action.stop();
+
     if (flags.json) {
```
Comment (RodEsp) [inline src/commands/login/functions/jwt.ts:213]: Same, no `if` statement needed.

File: src/commands/logout/functions.ts [hunk 11]
```diff
@@ -30,12 +30,9 @@ export default class Login extends Command {
     await this.stateAggregator.tokens.write();
 
     cli.action.stop();
+
     if (flags.json) {
```
Comment (RodEsp) [inline src/commands/logout/functions.ts:34]: Same, no `if` statement needed.

**Review by RodEsp — CHANGES_REQUESTED** [review]

File: src/commands/deploy/functions.ts (hunk 1 above)
Comment (RasPhilCo) [reply src/commands/deploy/functions.ts:194]: Interesting. Ok!

File: src/commands/env/create/compute.ts [hunk 12]
```diff
@@ -64,9 +63,7 @@ export default class EnvCreateCompute extends Command {
           )} to the "features" list in your scratch org definition JSON file, e.g. "features": ["Functions"]`
       );
     }
-    if (!flags.json) {
-      cli.action.start(`Creating compute environment for org ID ${orgId}`);
-    }
```
Comment (RodEsp) [inline src/commands/env/create/compute.ts:69]: Did you QA this?
I'm not sure that `cli.action.start()` has the same `--json` checks that `this.log()` does.

File: src/commands/env/logdrain/remove.ts [hunk 13]
```diff
@@ -67,37 +66,31 @@ export default class LogDrainRemove extends Command {
...
-        return;
-      } else {
-        cli.action.start(`Deleting drain for environment ${herokuColor.app(targetCompute)}`);
-
-        cli.action.stop();
-      }
+
+      cli.action.stop();
+
+      return [];
```
Comment (RodEsp) [inline src/commands/env/logdrain/remove.ts:84]: I won't hold up this PR for this but you should return _something_.
For these types of commands we often return the thing/object that was removed with a success message.

File: src/commands/env/var/set.ts [hunk 14]
```diff
@@ -118,42 +68,33 @@ export default class ConfigSet extends Command {
...
+    );
 
+    try {
       await this.client.patch(`/apps/${appName}/config-vars`, {
         data: configPairs,
       });
 
       cli.action.stop();
+
+      return [];
```
Comment (RodEsp) [inline src/commands/env/var/set.ts:91]: I won't hold up this PR for this but you should return _something_.
For these types of commands we often return the thing/object that was set.

File: src/commands/env/var/unset.ts [hunk 15]
```diff
@@ -92,26 +91,17 @@ export default class ConfigUnset extends Command {
...
-    }
-
-    if (flags.json) {
-      cli.styledJSON({
-        status: 0,
-        result: null,
-        warnings: [],
-      });
-    }
+    return [];
```
Comment (RodEsp) [inline src/commands/env/var/unset.ts:105]: I won't hold up this PR for this but you should return _something_.
For these types of commands we often return the thing/object that was unset.

File: src/commands/logout/functions.ts [hunk 16]
```diff
@@ -32,12 +30,7 @@ export default class Login extends Command {
...
     cli.action.stop();
-    if (flags.json) {
-      cli.styledJSON({
-        status: 0,
-        result: [],
-        warnings: [],
-      });
-    }
+
+    return [];
```
Comment (RodEsp) [inline src/commands/logout/functions.ts:34]: I won't hold up this PR for this but you should return _something_.
For these types of commands we often return the env that was logged out.

**Review by RodEsp — APPROVED** [review]
This all looks good to me but I'm not setup to run commands against an org with functions enabled so I can't QA. Trusting you guys are 😄

File: src/commands/env/create/compute.ts (hunk 12 above)
Comment (RasPhilCo) [reply src/commands/env/create/compute.ts:69]: Actions go out on stderr, so this shouldn't matter.

Comment (RasPhilCo) [reply src/commands/env/create/compute.ts:69]: _"He says 99% sure."_

File: src/commands/env/logdrain/remove.ts (hunk 13 above)
Comment (RasPhilCo) [reply src/commands/env/logdrain/remove.ts:84]: 10-4

File: src/commands/env/create/compute.ts (hunk 12 above)
Comment (RasPhilCo) [reply src/commands/env/create/compute.ts:69]: https://github.com/oclif/cli-ux/blob/21e8525497e6b4e9b4fa6cc1e2006bae2ff402eb/src/action/base.ts#L29

**Pull request merged**